### PR TITLE
allow $:/core/ui/ControlPanel/Basics to work in the story river + some docs about utility classes

### PR DESCRIPTION
**Pull request by pmario**
## For reviewers

See: https://pmario.github.io/kitchensink/6912-table-utility-classes.html

- **Please check the wording in the docs text** 

## To reproduce: 

- open https://tiddlywiki.com/#%24%3A%2Fcore%2Fui%2FControlPanel%2FBasics 
- the text input elements don't have the right size
- The second problem with tables in general. `code snippets` clash with table borders

<details open><summary>Screenshot -- Problem with ControlPanel table in story river</summary>

image

</details>

The second problem with tables in general

<details open><summary>Screenshot -- Code snippets in tables clash with table border</summary>

image

</details>

## This PR

- This PR allows the $:/core/ui/ControlPanel/Basics tiddler to work in the story river. 
- The screenshot also shows the Basics tiddler in the sidebar, to prove that the used classes also work there. 
- The PR also contains 3 doc tiddlers that have been updated with the info needed to understand the concept
- This PR should also deal with the second `code snippet` problem described above. 

- The tiddler:  editions/tw5.com/tiddlers/wikitext/Tables in WikiText Utility Classes.tid  contains a lot of whitespace, that will be removed. **It only makes editing with the preview ON easier.**

## Tested with 

- FF 105.03 Windows 11
- Edge Version 106.0.1370.37 Win 11
- FF 105.2.0 on Android 13
- Chrome on Android 13

This PR should also allow users to create table based "input forms" similar to those we use in the ControlPanel. IMO user forms should be consistent with the current theme. 

image

### CODE REVIEW

**Comment by pmario** [comment]
@twMat ... Could you please have a look at the docs?

**Comment by twMat** [comment]
> @twMat ... Could you please have a look at the docs?

The above linked zip file, right?
Will do, hopefully already tonight.
Can I give the feedback by replying with a modified zip?

**Comment by pmario** [comment]
>Can I give the feedback by replying with a modified zip?

sure

**Comment by twMat** [comment]
[basics-fix-in-story-river-plus-docs_twMat.zip](https://github.com/Jermolene/TiddlyWiki5/files/9377499/basics-fix-in-story-river-plus-docs_twMat.zip)

OK, I've fiddled around in ~4 doc tiddlers. Hope this is what you requested.
Further:

* I propose calling them "Utility Styles" rather than "Utility Classes". Yes, I understand that technically it is more correct with "Classes" but it is unfortunately ambiguous whereas "Styles" is directly clear for both "CSS people" *and* "non-CSS people".
* I propose simply calling it "Tables" rather than "Tables in WikiText", at least in the headings, because it is understood the docs are about WikiText.

...Thus the tiddler title could be something like "Tables and Utility Styles" or even just "Styling tables" or "Table styling"

* OT: I think we should have a separate doc tiddler on CSS that we link to any time CSS is mentioned. Beyond a brief general note about what CSS is, the doc should explain CSS' role in TW and link to where it is brought up, but also make it clear that it is mostly outside the scope of the TW docs.

**Comment by pmario** [comment]
@twMat ... It seems you did modify the quote in the "HTML Block Elements" tiddler ... IMO that's not possible. Quotes have to be exact and shouldn't be modified, even if they have typos. 

It's possible to crate a PR directly at the mozilla repo, that is linked at the bottom of the page: https://developer.mozilla.org/en-US/docs/Web/HTML/Block-level_elements. 

They provide a link: [Edit on GitHub](https://github.com/mdn/content/edit/main/files/en-us/web/html/block-level_elements/index.md) ... 

You should use this one and fix the problem there. .. They are very kind folks and they are happy to get feedback. I did create several issues, with some problems I couldn't fix, but I wanted to let them know. They have been fixed in the meantime.

**Comment by twMat** [comment]
> It seems you did modify the quote in the "HTML Block Elements" tiddler

Ah, I didn't realize it was a direct quote but, regardless, I don't think I actually changed anything in the end. Anyway, just disregard that particular tiddler and I hope the other mods made sense :)

File: editions/tw5.com/tiddlers/Tables in WikiText CSS Utility Classes.tid [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+modified: 20220819103351008
+tags: [[Tables in WikiText]]
+title: Tables in WikiText CSS Utility Classes
+type: text/vnd.tiddlywiki
+
+~WikiText tables can be styled by applying CSS classes. For basic information on wiktext tables and how to style them, see [[Tables in WikiText]].
+
+{{Table Classes, Captions, Headers and Footers}}
+
+As seen above, the resulting table that is left aligned and grows to fit the content. This is the browser default layout behaviour for tables. To get another behaviour, various CSS classes can be added into the "`k` row".
```
Comment (twMat) [inline editions/tw5.com/tiddlers/Tables in WikiText CSS Utility Classes.tid:11]: As seen above, the resulting table XXXthatXXX is left aligned and

File: editions/tw5.com/tiddlers/Tables in WikiText CSS Utility Classes.tid [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+
+
+
+
+
+
+! Examples
+
+The following examples apply the style classes WikiText tables but the General Utility Classes can as well be used on a DIV or other [[HTML block elements|HTML Block Elements]]
```
Comment (twMat) [inline editions/tw5.com/tiddlers/Tables in WikiText CSS Utility Classes.tid:65]: maybe it should say "...WikiText TO tables but..."

File: editions/tw5.com/tiddlers/wikitext/Utility Classes.tid [hunk 3]
```diff
@@ -1,27 +1,21 @@
...
-!! Table Utility Class
-
-* `tc-table-no-border` ... removes the border of a table
-
-* `tc-first-col-min-width` ... The first column of a table will consume the minimal size. It adapts to the content
-
-Examples how to use generic and table specific utility classes can be found at [[Tables in WikiText Utility Classes]].
\ No newline at end of file
+|`tc-table-no-border` |Removes the borders of a table |
+|`tc-first-col-min-width` |The first column of a table will consume the minimal size. It adapts to the content |
```
Comment (twMat) [inline editions/tw5.com/tiddlers/wikitext/Utility Classes.tid:21]: Maybe "take up minimal width" instead of "consume the minimal size" ?

**Comment by pmario** [comment]
>... I hope the other mods made sense :)

Absolutely. ... I think with this in place, there can be a lot more docs about "user created forms" that look nice out of the box. 

We have a lot of CSS related requests at Talk, where from my point of view, the "solutions" sometimes are really complex CSS constructions. .. But that's not the users fault. It's just the lack of knowledge and some very basic core definitions. + some docs ;)

I think, they will not only help users, but also bring better consistency to the core UI itself. Even if the differences are almost invisible, like a little bit of padding added to the default tables. .. In the end it makes a big difference in readability.

**Comment by pmario** [comment]
@Jermolene .. I did update the ZIP file to the latest version 

There is still a lot of whitespace in 1 tiddler for easier editing ... But I think it should be reviewed. 

If you are OK with it I'll remove the whitespace and create a "real PR"

**Comment by pmario** [comment]
I did test it with FF on windows and FF on my Android phone .. Looks good an the phone too

**Comment by pmario** [comment]
@Jermolene .. Ready for review again -- I did update the OP with a new link to the test version: See: https://pmario.github.io/kitchensink/6912-table-utility-classes.html

It looks good with all the browsers I do have available. See updated OP

**Comment by pmario** [comment]
@Jermolene pls merge or comment

**Comment by Jermolene** [comment]
Hi @pmario I've reverted this PR in the light of #7037, are you able to rework it?

**Comment by Jermolene** [comment]
@pmario I'm afraid I still have a problem with these changes, and it's an issue we have discussed before.

I think it is a mistake to overwrite browser styles for built-in HTML controls. For example, these recent changes make the text boxes in control panel look the same in all browsers. That really should not be a goal for us. We get better compatibility and usability by using operating system default styling.

I will consider whether to pull those particular changes. It would be good to keep the other enhancements though.

**Comment by pmario** [comment]
I'm pretty sure, that this PR doesn't change anything with the edit input elements.

**Comment by pmario** [comment]
@Jermolene The PR https://github.com/Jermolene/TiddlyWiki5/pull/6910 changed those settings because there has been a problem with the new reset we used starting with 5.1.23. It needed to fix the focus-ring for FF. 

We did have a lot of requests at Talk that the UI should look the same for all browsers.

**Pull request merged**